src/app/return-credentials/tds-return/tds-return.component.ts

```diff
@@ -9,23 +12,41 @@ import {FormGroup, FormBuilder} from '@angular/forms';
 })
 export class TdsReturnComponent implements OnInit {
 
+  private clientId: string;
   private tdsReturnForm: FormGroup;
 
-  constructor(private formBuilder: FormBuilder, private apiService: SyncupApiService) {
-    this.tdsReturnForm = formBuilder.group({
-                    tdsTanNo: [''],
-                    tdsUserName: [''],
-                    rocPassword: [''],
-                    tracesTdsUserName: [''],
-                    tracesTdsPassword: ['']
+  constructor(private formBuilder: FormBuilder, private apiService: SyncupApiService, private dataTransferService: DataTransferService, private router: Router) {
+    this.tdsReturnForm = this.formBuilder.group({
+                    tdsTanNo: [null, [Validators.required]],
+                    tdsUserName: [null, [Validators.required]],
+                    tdsPassword: [null, [Validators.required]],
+                    tracesTdsUserName: [null, [Validators.required]],
+                    tracesTdsPassword: [null, [Validators.required]]
                   });
    }
 
   ngOnInit() {
+    this.dataTransferService.currentMessage.subscribe(message => this.clientId = message);
   }
 
   private addTdsReturnInfo() {
     console.log(this.tdsReturnForm.value);
+    const tdsReturnCredentials: ReturnCredentials = new ReturnCredentials();
+    tdsReturnCredentials.setUserId = this.tdsReturnForm.controls.tdsUserName.value;
```

Comment:
what in case of null ? ek baar see usage of idx

Reply:
idx matlab